### PR TITLE
move frontend code into a folder, and setup react build tooling with es6 transforms

### PR DESCRIPTION
**Pull request by devvmh**
Take a look, let me know what you think.

### CODE REVIEW

File: frontend/src/components/ExploreHeader.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React, { Component, PropTypes } from 'react'
```
Comment (devvmh) [inline frontend/src/components/ExploreHeader.js]: These imports could be changed to requirejs format if we want to

File: frontend/src/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+// this is optional really, if we import components directly React will be 
+// in the bundle, so we won't need a global reference
+window.React = React
+window.ReactDOM = ReactDOM
+
+window.Metamaps = window.Metamaps || {}
+window.Metamaps.ReactComponents = {
+  ExploreHeader
+}
```
Comment (devvmh) [inline frontend/src/index.js]: I envision either having a bunch of components available for use in the javascript files over in app/assets/javascripts, or this file could call ReactDOM.render a bunch of times.

File: frontend/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    ]
+  },
+  entry: {
+    'metamaps.bundle': './src/index.js'
+  },
+  output: {
+    path: '../app/assets/javascripts/webpacked',
+    filename: '[name].js'
+  }
+}
```
Comment (devvmh) [inline frontend/webpack.config.js]: This file just basically sets things up so running `webpack` builds `src/index.js` and all imported modules into a bundle at `../app/assets/javascripts/webpacked/metamaps.bundle.js`

File: frontend/.babelrc [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "presets": [
+    "react",
+    "es2015"
+  ],
+  "plugins": [
+    "transform-class-properties"
+  ]
+}
```
Comment (devvmh) [inline frontend/.babelrc]: This file just sets up babeljs to handle react and es6 code. I was hoping to have no babel plugins (and just pure es6), but react code is much more natural to write with transform-class-properties so I bit my tongue.

without transform-class-properties, the component below would need to have a constructor and call `.bind(this)` on every function (including render) to be able to access the react props... it's just dumb. It's much simpler to use arrow functions for every function in the react component, and this works exactly how you want it to.

**Comment by Connoropolous** [comment]
Beautiful Devin. What I'm not sure about is... 

would the best strategy for now be building a bunch of components, and injecting them individually into a number of different divs on the page, or would it better to write one 'app' that manages any react components we have so far?

**Comment by Connoropolous** [comment]
also, to start using this code we'd need to require into application.js or directly import it into the html file yeah?

File: frontend/.babelrc (hunk 4 above)
Comment (harlantwood) [inline frontend/.babelrc]: Nice.  Didn't know about `transform-class-properties`.  Thanks!

**Comment by harlantwood** [comment]
Excited about this!  Great work.

File: frontend/package.json [hunk 5]
```diff
@@ -16,7 +18,18 @@
...
   "devDependencies": {
+    "babel-cli": "^6.11.4",
+    "babel-loader": "^6.2.4",
+    "babel-plugin-transform-class-properties": "^6.11.5",
+    "babel-preset-es2015": "^6.9.0",
+    "babel-preset-react": "^6.11.1",
     "chai": "^3.5.0",
-    "mocha": "^2.4.5"
+    "mocha": "^2.4.5",
+    "path": "^0.12.7",
```
Comment (harlantwood) [inline frontend/package.json]: You get `path` for free from node.js, no need to put it in package.json

File: frontend/src/index.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import ExploreHeader from './components/ExploreHeader.js'
```
Comment (harlantwood) [inline frontend/src/index.js]: .js is optional

**Comment by Connoropolous** [comment]
thanks for the input and encouragement @harlantwood :)

**Pull request merged**